AccessibilitySnapshot/Classes/FBSnapshotTestCase+ObjCSupport.swift

```diff
@@ -49,6 +49,14 @@ extension FBSnapshotTestCase {
         activationPointDisplayMode: ActivationPointDisplayMode,
         useMonochromeSnapshot: Bool
     ) -> String? {
+        // The tests must be run in a host application in order for the accessibility properties to be populated
+        // correctly. The `UIApplication.shared` singleton is non-optional, but will be uninitialized when the tests are
+        // running outside of a host application, so we can use this check to determine whether we have a test host.
+        let hostApplication: UIApplication? = UIApplication.shared
+        guard hostApplication != nil else {
+            return "Accessibility snapshot tests cannot be run in a test target without a host application"
```

Comment:
Asking to learn: why are we returning an error message here? Seems like a reasonable error message but I’m having trouble understanding why it’s being returned as a string.

Reply:
This is how the Objective-C test assertions put the error on the correct line. The actual test code is a method that returns a `String?`. The Objective-C tests use a macro that calls this method. If the method returns nil, it considers the test to have succeeded. If the method returns a string, it considers the test to have failed and calls `XCTFail()` with that as the failure message. Since it's a macro, the failure shows up on that line, rather than in the framework code.

Reply:
Got it! Thanks for the explanation.